Fauna-RPG/Entity/Components/HealthComponent.gd

```diff
@@ -0,0 +1,72 @@
+extends EntityComponent
+class_name HealthComponent
+## Holds player health values and allows for dealing damage.
+
+
+signal on_health_changed(new_health : int)
+
+@export var max_health : float = 100
+var current_health : float = 100
+
+
+func _ready() -> void:
+	_set_health(max_health)
+
+
+# Allow attacker to be null, but not the victim.
+# If attacker is null, we assume that the damage came from an unknown source,
+# so buffs won't calculate.
+# Use a negative damage value if you want to heal.
```

Comment:
we should make a function to heal which just calls the damage function with a negative number to not get confused later